ServerLayer/Source/ServerLayer.h

```diff
@@ -1,287 +1,271 @@
-
-	/**
-	 * \brief					Set server IP address.
-	 * \param[in] newServerIP	New server IP address as string.
-	 */
-	void		    setServerIP(const std::string_view newServerIP);
-
-	/**
-	 * \brief Additional fuction that receives data from server.
-	 */
-	void		    receiveFromServer();
-
-	/**
-	 * \brief Additional fuction that receives data from clients.
-	 */
-	void            receiveFromClients();
-
-	/**
-	 * \brief				Check if given point is not out of working coordinates.
-	 * \param[in] robotData	Point to check.
-	 * \return				True if point is correct, false otherwise.
-	 */
-	bool		checkCoordinates(const RobotData& robotData) const;
+    /**
+     * \brief Array of modes for layer how to process data.
+     */
+    enum class WorkMode
+    {
+        SAFE,
+        UNSAFE
+    };
+
+    /**
+     * \brief Array of constant to get parameters from config.
+     */
+    enum class Param : std::size_t
+    {
+        DEFAULT_IN_FILE_NAME,
+        DEFAULT_OUT_FILE_NAME,
+        DEFAULT_SERVER_IP,
+        DEFAULT_SENDING_PORT_TO_SERVER,
+        DEFAULT_RECEIVING_PORT_FROM_SERVER,
+        DEFAULT_CLIENT_PORT,
+        NUMBER_OF_MAIN_COORDINATES,
+        MIN_COORDINATES,
+        MAX_COORDINATES,
+        RECONNECTION_DELAY
+    };
+
+    /**
+     * \brief   Variable used to keep all default parameters and constants.
+     * \details Using std::string instead of std::string_view because Logger constructor needs only
+     *          std::string because of std::istream and std::ostream.
+     */
+    static const config::Config<std::string, std::string, std::string_view, int, int, int,
+                                std::size_t, std::array<int, 3u>, std::array<int, 3u>, long long>
+        CONFIG;
+
+    /**
+     * \brief                        Constructor that initializes sockets, connects to server and
+     *                               creates socket to work with clients.
+     * \param[in] serverSendingPort  Server port to send.
+     * \param[in] serverRecivingPort Server port to recieve.
+     * \param[in] serverIP           Server IP address for connection.
+     * \param[in] layerPort          Additional port to communicate with clients.
+     * \param[in] backlog            The maximum length of the queue of pending connections.
+     * \param[in] workMode           Set work mode for layer to work with robot in safety
+     *                               or unsafety.
+     */
+    explicit ServerLayer(
+        const int serverSendingPort        = CONFIG.get<CAST(Param::DEFAULT_SENDING_PORT_TO_SERVER)>(),
+        const int serverRecivingPort       = CONFIG.get<CAST(Param::DEFAULT_RECEIVING_PORT_FROM_SERVER)>(),
+        const std::string_view serverIP    = CONFIG.get<CAST(Param::DEFAULT_SERVER_IP)>(),
+        const int layerPort                = CONFIG.get<CAST(Param::DEFAULT_CLIENT_PORT)>(),
+        const int backlog                  = 10,
+        const WorkMode workMode            = WorkMode::SAFE);
```

Comment:
**(note):** why you use hardhode here and don't pull up parameters "backlog" and "workMode"(especially it) from file?

Reply:
WordkMode and backlog are not so impportant parameters for classes. WorkMode is needed only to separate debug version and release version of using class with this parameter. Backlog is binding to WinSock and will be removed in Qt version of our prooject.